search/v2/geocoder.cpp

```diff
@@ -609,6 +659,44 @@ void Geocoder::FillLocalitiesTable(MwmContext const & context)
   }
 }
 
+void Geocoder::FillVillageLocalities()
+{
+  vector<Locality> preLocalities;
+  FillLocalityCandidates(m_villages, kMaxNumVillages, preLocalities);
+
+  size_t numVillages = 0;
+
+  for (auto & l : preLocalities)
+  {
+    FeatureType ft;
+    m_context->m_vector.GetByIndex(l.m_featureId, ft);
+
+    switch (m_model.GetSearchType(ft))
+    {
+    case SearchModel::SEARCH_TYPE_CITY:
```

Comment:
There are no cities in m_villages, hence there are no cities in preLocalities.
Still, I'll think about it.